src/pybind/mgr/nfs/export.py

```diff
@@ -186,6 +187,27 @@ def nfs_rados_configs(rados: 'Rados', nfs_pool: str = POOL_NAME) -> Set[str]:
     return ns
 
 
+class AppliedExportResults:
+    """Gathers the results of multiple changed exports.
+    Returned by apply_export.
+    """
+
+    def __init__(self) -> None:
+        self.changes: List[Dict[str, str]] = []
+        self.has_error = False
+
+    def append(self, value: Dict[str, str]) -> None:
+        if value.get("state", "") == "error":
+            self.has_error = True
+        self.changes.append(value)
+
+    def to_simplified(self) -> List[Dict[str, str]]:
+        return self.changes
+
+    def mgr_return_value(self) -> int:
+        return -errno.EIO if self.has_error else 0
```

Comment:
Interesting. Previously,  if exceptions were raised by _apply_export() calls,  the return value was getattr(exception_obj, 'errno', -1), where exception_obj was the latest exception raised by a _apply_export() call. 

We're now always setting the return code to -errno.EIO  if any exception is raised when applying the exports. Because EIO is the most generic error?

Reply:
Funny, I don't remember changing this on purpose. :-)  Anyway, I think that may be the case -1 is -EPERM, which probably didn't make sense as a general error.  However, this is a case where I don't have strong feelings about it.

Reply:
I like your change as is. I wanted to better understand your rationale.